File: 2448-count-number-of-bad-pairs/2448-count-number-of-bad-pairs.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var countBadPairs = function (a) {
    const { length } = a;
    const m = new Map();
    let p = length * (length - 1) / 2;
    for (let i = 0, d = a[0]; i < a.length; d = a[++i] - i) {
        const t = m.get(d) ?? 0;
        p -= t;
        m.set(d, t + 1);
    }
    return p;
};

File: 2448-count-number-of-bad-pairs/2448-count-number-of-bad-pairs.cs
public class Solution {
    public long CountBadPairs(int[] nums) {
        var n = nums.Length;
        var map = new Dictionary<long, long>();
        for (long i = 0; i < n; i++) {
            long dif = nums[i] - i;
            if (map.ContainsKey(dif))
                map[dif]++;
            else
                map[dif] = 1;
        }
        long tot = (long )n * (n - 1) / 2;
        long good = 0;
        foreach(var item in map) {
            var val = item.Value;
            if (val > 0)
                good += val * (val - 1) / 2;
        }
        return tot - good;
    }
}